src/main/java/com/conveyal/datatools/manager/models/OtpServer.java

```diff
@@ -91,4 +125,313 @@ public Project parentProject() {
+    private EC2ValidationResult validateAmiId(String amiId) {
+        EC2ValidationResult result = new EC2ValidationResult();
+        if (isEmpty(amiId)) return result;
+        try {
+            if (!EC2Utils.amiExists(getEC2Client(), amiId)) {
+                result.setInvalid("Server must have valid AMI ID (or field must be empty)");
+            }
+        } catch (AmazonEC2Exception | CheckedAWSException e) {
+            result.setInvalid("AMI does not exist or some error prevented proper checking of the AMI ID.", e);
+        }
+        return result;
+    }
+
+    /**
+     * Validates whether the replacement graph build image name is unique. Although it is possible to have duplicate AMI
+     * names when copying images, they must be unique when creating images.
+     * See https://forums.aws.amazon.com/message.jspa?messageID=845159
+     */
+    @JsonIgnore
+    @BsonIgnore
+    private EC2ValidationResult validateGraphBuildReplacementAmiName() {
+        EC2ValidationResult result = new EC2ValidationResult();
+        if (!ec2Info.recreateBuildImage) return result;
+        String buildImageName = ec2Info.buildImageName;
+        try {
+            DescribeImagesRequest describeImagesRequest = new DescribeImagesRequest()
+                // limit AMIs to only those owned by the current ec2 user.
+                .withOwners("self");
+            DescribeImagesResult describeImagesResult = getEC2Client().describeImages(describeImagesRequest);
+            // Iterate over AMIs to see if any images have a duplicate name.
+            for (Image image : describeImagesResult.getImages()) {
+                if (image.getName().equals(buildImageName)) {
+                    result.setInvalid(String.format("An image with the name `%s` already exists!", buildImageName));
+                    break;
+                }
+            }
+        } catch (AmazonEC2Exception | CheckedAWSException e) {
+            String message = "Some error prevented proper checking of for duplicate AMI names.";
+            LOG.error(message, e);
+            result.setInvalid(message, e);
+        }
+        return result;
+    }
+
+    /** Validate that AWS key name (the first part of a .pem key) exists and is not empty. */
+    @JsonIgnore
+    @BsonIgnore
+    private EC2ValidationResult validateKeyName() throws CheckedAWSException {
+        String message = "Server must have valid key name";
+        EC2ValidationResult result = new EC2ValidationResult();
+        if (isEmpty(ec2Info.keyName)) {
+            result.setInvalid(message);
+            return result;
+        };
+        DescribeKeyPairsResult response = this.getEC2Client().describeKeyPairs();
+        for (KeyPairInfo key_pair : response.getKeyPairs()) {
+            if (key_pair.getKeyName().equals(ec2Info.keyName)) return result;
+        }
+        result.setInvalid(message);
+        return result;
+    }
+
+    /** Validate IAM instance profile ARN exists and is not empty. */
```

Comment:
"Validate **that** ..."

Reply:
Fixed in 5d3a85bad9f911fbc3f83fccd80bb476f5a5f511